[tests] update CI to use node@14 and drop support for node@12

EndangeredMassa opened the pull request:
Node 12 is EOL and we're starting to reference packages that don't install on node lower than 14, such as nuxt@3 and remix. Let's update to 14 for now.

EndangeredMassa committed:
update CI to use node@14

styfle reviewed (commented):
We should also update tsconfig.json target and package.json engines, then we can ship semver major to drop support for 12

EndangeredMassa commented:
@TooTallNate we'll need to change the "Required" checks. We should remove the node@12 versions, add the node@14 versions, and probably also add the new matrix of checks, right?

EndangeredMassa committed:
Merge branch 'main' into endangeredmassa/update-ci-node-14

EndangeredMassa committed:
update tsconfig throughout

EndangeredMassa committed:
update engines to node 14

styfle commented on packages/cli/package.json:
```diff
@@ -40,7 +40,7 @@
   },
   "engines": {
-    "node": ">= 12"
+    "node": "14.x"
```
```suggestion
    "node": ">= 14"
```

We don't want to prevent users from using Node.js 16

TooTallNate commented on packages/client/package.json:
```diff
@@ -20,7 +20,7 @@
   },
   "engines": {
-    "node": ">= 12"
+    "node": "14.x"
```
```suggestion
    "node": ">= 14"
```

TooTallNate committed:
Merge branch 'main' into endangeredmassa/update-ci-node-14

EndangeredMassa committed:
Update packages/cli/package.json

Co-authored-by: Steven <steven@ceriously.com>

EndangeredMassa committed:
Update packages/client/package.json

Co-authored-by: Nathan Rajlich <n@n8.io>

EndangeredMassa committed:
Merge branch 'main' into endangeredmassa/update-ci-node-14

EndangeredMassa committed:
Merge branch 'main' into endangeredmassa/update-ci-node-14

EndangeredMassa committed:
Merge remote-tracking branch 'origin' into endangeredmassa/update-ci-node-14

EndangeredMassa committed:
Merge branch 'main' into endangeredmassa/update-ci-node-14

EndangeredMassa committed:
Merge branch 'main' into endangeredmassa/update-ci-node-14

EndangeredMassa committed:
Merge branch 'main' into endangeredmassa/update-ci-node-14

EndangeredMassa commented:
I updated the tsconfig files based on the [Node Target Mapping](https://github.com/microsoft/TypeScript/wiki/Node-Target-Mapping) guidance for typescript. I changed every `target` and `lib` key that I found to `ES2020`, even if they were `esnext` before. I think this makes the most sense, but if `esnext` is a better choice for these, please let me know why and I'll change them.

I left the examples alone.

TooTallNate commented:
`next` and `remix` packages may need to be updated as well

EndangeredMassa committed:
update tsconfig for remix and next

EndangeredMassa commented:
@TooTallNate good call. Updated!

EndangeredMassa committed:
Merge branch 'main' into endangeredmassa/update-ci-node-14

EndangeredMassa committed:
Merge branch 'main' into endangeredmassa/update-ci-node-14

EndangeredMassa committed:
Merge branch 'main' into endangeredmassa/update-ci-node-14

styfle commented on packages/node/src/typescript.ts:
```diff
@@ -450,17 +450,14 @@ export function fixConfig(
     if (nodeVersionMajor >= 16) {
       target = 'ES2021';
-    } else if (nodeVersionMajor >= 14) {
-      target = 'ES2020';
```
We need to leave this until August 9th

https://vercel.com/changelog/node-js-12-is-being-deprecated

EndangeredMassa committed:
enhance error message from now-deploy

EndangeredMassa committed:
Merge branch 'main' into endangeredmassa/update-ci-node-14

EndangeredMassa committed:
restore node version cascade

EndangeredMassa committed:
tweak return values of now-deploy
